### PR TITLE
host: Update browser to show active file upon render

### PR DESCRIPTION
**Pull request by backspace**
This is a small UX improvement for #585, as one would expect the selected file to be visible when the file tree view is chosen, see it [here](https://hostview-active-file-cs-5964.boxel-host-preview.stack.cards/?operatorModeEnabled=true&operatorModeState=%7B%22codePath%22%3A%22https%3A%2F%2Frealms-staging.stack.cards%2Fdrafts%2FFriend%2Fdog%2F2.json%22%2C%22fileView%22%3A%22browser%22%2C%22openDirs%22%3A%5B%22Friend%2Fdog%2F%22%5D%2C%22stacks%22%3A%5B%5B%7B%22format%22%3A%22isolated%22%2C%22id%22%3A%22https%3A%2F%2Frealms-staging.stack.cards%2Fdrafts%2Findex%22%7D%2C%7B%22format%22%3A%22isolated%22%2C%22id%22%3A%22https%3A%2F%2Frealms-staging.stack.cards%2Fdrafts%2FFriend%2F6%22%7D%5D%5D%2C%22submode%22%3A%22code%22%7D). I also included a bug fix where multiple files could appear as active if one was a post-fix of the other, I’ve commented directly on the relevant code.

This changes `OperatorModeStateService#updateCodePath` to add a directory to `openDirs` if the new path is within a directory that’s not currently open.

It does not attempt to preserve scroll position when switching between file views, which is recorded in 6033.

### CODE REVIEW

File: packages/host/tests/acceptance/code-mode-test.ts [hunk 1]
```diff
@@ -282,12 +284,13 @@ module('Acceptance | code mode tests', function (hooks) {
...
+    await waitFor('[data-test-file="pet-person.gts"]');
 
-    await click('[data-test-file="person.gts"]');
+    await click('[data-test-file="pet-person.gts"]');
 
-    await waitFor('[data-test-file="person.gts"]');
-    assert.dom('[data-test-file="person.gts"]').hasClass('selected');
+    await waitFor('[data-test-file="pet-person.gts"]');
+    assert.dom('[data-test-file="pet-person.gts"]').hasClass('selected');
+    assert.dom('[data-test-file="person.gts"]').doesNotHaveClass('selected');
```
Comment (backspace) [inline packages/host/tests/acceptance/code-mode-test.ts:293]: This is asserting against this bug:

Boxel 2023-09-15 13-22-59

File: packages/host/app/components/editor/directory.gts [hunk 2]
```diff
@@ -144,14 +164,57 @@ function fileIsSelected(
   localPath: string,
   operatorModeStateService: OperatorModeStateService,
 ) {
-  return operatorModeStateService.state.codePath?.pathname.endsWith(localPath);
+  return operatorModeStateService.codePathRelativeToRealm === localPath;
```
Comment (backspace) [inline packages/host/app/components/editor/directory.gts:151]: The use of `endsWith` is what caused the multi-selected bug.

File: packages/host/app/components/editor/directory.gts [hunk 3]
```diff
@@ -113,6 +117,18 @@ export default class Directory extends Component<Args> {
...
+        color: var(--boxel-dark);
+      }
+
+      .directory[disabled]:hover {
+        background: transparent;
+      }
+
+      .directory[disabled] :deep(.icon) {
+        --icon-color: var(--boxel-light-600);
+      }
```
Comment (jurgenwerk) [inline packages/host/app/components/editor/directory.gts]: Are these still needed?

**Review by jurgenwerk — APPROVED** [review]
I tested it and I like how it works. Nice!

File: packages/host/app/components/editor/directory.gts (hunk 3 above)
Comment (backspace) [reply packages/host/app/components/editor/directory.gts]: they are indeed abandoned, removed now, thanks … though the push is not showing here 🧐

**Pull request merged**